nautobot/ipam/factory.py

```diff
@@ -54,6 +72,45 @@ class Meta:
 
     is_ipv6 = factory.Faker("pybool")
 
+    @factory.post_generation
+    def child_prefixes(self, create, extracted, **kwargs):
+        """
+        Create child prefixes within the aggregate IP space. Defaults to generating 0-4 child
+        prefixes for 50% of aggregates. Set child_prefixes__max_count to an integer when calling
+        the factory creation methods (`create()`, `create_batch()`, etc) to override the maximum
+        number of children generated.
+        """
+        if extracted:
+            # Objects have already been created, do nothing
+            return
+
+        # 50% chance to create child prefixes
+        if not faker.Faker().pybool():
+            return
+
+        action = "create" if create else "build"
+        method = getattr(PrefixFactory, action)
+        is_ipv6 = self.family == 6
+
+        # Default to maximum of 4 children unless overridden in kwargs
+        max_count = int(kwargs.pop("max_count", 4))
+        prefix_count = faker.Faker().pyint(min_value=0, max_value=min(max_count, self.prefix.size))
+        if prefix_count == 0:
+            return
+
+        # Calculate prefix length for child prefixes to allow them to fit in the aggregate
+        prefix_cidr = self.prefix_length + math.ceil(math.log(prefix_count, 2))
+
+        # Raise exception for invalid cidr length (>128 for ipv6, >32 for ipv4)
+        if prefix_cidr > 128 or self.family == 4 and prefix_cidr > 32:
+            raise ValueError(f"Unable to create {prefix_count} prefixes in aggregate {self.cidr_str}")
+
+        # Create child prefixes, preserving tenant and is_ipv6 from aggregate
+        for count, subnet in enumerate(self.prefix.subnet(prefix_cidr)):
+            if count == max_count:
+                break
+            method(prefix=str(subnet.cidr), tenant=self.tenant, is_ipv6=is_ipv6, **kwargs)
```

Comment:
Since Aggregate.tenant is optional, maybe allow prefixes of a non-tenanted Aggregate to have random tenants of their own? Something like:

```suggestion
            if self.tenant is not None:
                method(prefix=str(subnet.cidr), tenant=self.tenant, is_ipv6=is_ipv6, **kwargs)
            else:
                method(prefix=str(subnet.cidr), is_ipv6=is_ipv6, **kwargs)
```